### PR TITLE
Reorder chapters in server guide

### LINKED ISSUE #20354: Reorder chapters in server guide
### Description

As agreed in an internal discussion, they should have the following order:
1. Configuring Keycloak
2. Configuring Keycloak for production
3. Running Keycloak in a container
4. Configuring TLS
5. Configuring the hostname
6. Using a reverse proxy
7. Configuring the database
8. Configuring distributed caches
9. Configuring outgoing HTTP requests
10. Configuring trusted certificates for outgoing requests
11. Enabling and disabling features
12. Configuring providers
13. Configuring logging
14. FIPS 140-2 Support
15. Enabling Keycloak Health checks
16. Enabling Keycloak Metrics
17. Importing and Exporting Realms
18. Using Kubernetes secrets
19. All configuration
20. All provider configuration

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #20354

### CODE REVIEW

File: docs/guides/assembly.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
             </includes>
         </fileSet>
         <fileSet>
-            <directory>${project.basedir}/src/main/server</directory>
+            <directory>${project.basedir}/server</directory>
```
Comment (vmuzikar) [inline docs/guides/assembly.xml:16]: I sneaked in a little fix for keycloak-web. It turned out the pinned guides actually didn't work there.

CC: @ahus1

Comment (ahus1) [reply docs/guides/assembly.xml:16]: Ooops! Thanks for fixing this. Do we plan another KC minor release, and should this be backported?

Comment (vmuzikar) [reply docs/guides/assembly.xml:16]: I think nobody really noticed maybe. :) So perhaps no need for a backport?

CC @stianst

**Review by andymunro — APPROVED** [review]

**Comment by Pepo48** [comment]
LGTM as well.

**Review by ahus1 — APPROVED** [review]
LGTM, I've build the downstream docs archives and the upstream keycloak-web preview and they look fine.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**